codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -857,9 +866,16 @@ public Http2Stream closeRemoteSide() {
         }
 
         @Override
-        public int nextStreamId() {
-            // For manually created client-side streams, 1 is reserved for HTTP upgrade, so start at 3.
-            return nextStreamId > 1 ? nextStreamId : nextStreamId + 2;
+        public int incrementAndGetNextStreamId() {
```

Comment:
@Scottmitch I wonder if we can get rid of one of the variables.  If we changed this to `getAndIncrementNextStreamId`, then I think we would only need `nextStreamId`.  It would be initialized as it was before:

```java
nextStreamId = server ? 2 : 1;
```

The methods would then look something like this:

```java
@Override
public int getAndIncrementNextStreamId() {
    int value = nextStreamId;
    if (value == 1) {
        // For manually created client-side streams, 1 is reserved for HTTP upgrade, so start at 3.
        value += 2;
    }
    nextStreamId = value + 2;
    return value;
}

// Not really sure that this needs to be a method ... could probably just do this inline.
private void createStreamId(int streamId) {
    // We've already checked that streamId >= nextStreamId (via checkNewStreamAllowed).
    nextStreamId = streamId + 2;
    lastStreamCreated = streamId;
}
```

WDYT?